api/sip/client.py

```diff
@@ -230,37 +237,84 @@ def patron_information(self, *args, **kwargs):
             
     def connect(self):
         """Create a socket connection to a SIP server."""
-        try:
-            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        except socket.error, msg:
-            self.log.warn("Error initializing socket: %s", msg[1])
-
-        try:
-            sock.connect((self.target_server, self.target_port))
-        except socket.error, msg:
-            self.log.warn("Error connecting to SIP server: %s", msg)
+        with self.socket_lock:
+            try:
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            except socket.error, msg:
+                self.log.warn("Error initializing socket: %s", msg[1])
+
+            try:
+                sock.connect((self.target_server, self.target_port))
```

Comment:
Just noticed this will try to connect even if there was an error initializing the socket